zerver/lib/actions.py

```diff
@@ -1541,6 +1541,8 @@ def prep_stream_welcome_message(stream: Stream) -> Optional[Dict[str, Any]]:
 def send_stream_creation_event(stream: Stream, user_ids: List[int]) -> None:
     event = dict(type="stream", op="create",
                  streams=[stream.to_dict()])
+    # We add email_address to event, since Stream object passed here does not have one.
+    event['streams'][0]['email_address'] = encode_email_address(stream)
```

Comment:
I think it makes sense to modify a structure **before** putting it in something deeper, so I'd do something like this:

~~~ py
stream_dict = stream.to_dict()
stream_dict['email_address'] = encode_email_address(...)
event = dict(..., streams=[stream_dict])
~~~

This avoids the danger of accidentally associating the wrong email with the wrong stream.

We'll also want a backend test for this.

Reply:
Good idea, Thanks. and yes I'm currently taking a look at tests only!